### ISSUE TITLE
Don't register mouse press when restoring active window status to inactive game process

### ISSUE THREAD

**Issue by Kefta** [Facepunch/garrysmod-requests#968]
Essentially, if the game window is inactive/alt-tabbed out, don't count the mouse press upon making it active again if done by directly clicking back into the game. This will prevent a lot of those "accidental shot" situations when clicking back into the game.

**Comment by Tenrys** [comment]
Can be fixed with Lua but yeah

**Comment by iViscosity** [comment]
`if SERVER then
    util.AddNetworkString( "check_min" )
    function MarkPlayer( len, ply )
        ply:SetNWBool( "minimized", net.ReadBool() )
    end
    net.Receive( "check_min", MarkPlayer )
else
    function CheckPlayer()
        if not system.IsWindows() then return end -- system.HasFocus() doesn't work (correctly) on non-windows systems
        timer.Create( "check_for_min", 2, 0, function()
            net.Start( "check_min" )
                net.WriteBool( system.HasFocus() )
            net.SendToServer()
        end )
    end
end

function IgnoreClick( ply, mv )
    if not ply:GetNWBool( "minimized", false ) then return end -- No need to ignore if not minimized.
    if mv:KeyDown( IN_ATTACK ) or mv:KeyDown( IN_ATTACK2 ) then
        return true
    end
end
`


This should work. I didn't test it though, just a concept.

**Comment by Kefta** [comment]
Yeah, I'm aware it can be fixed with Lua -- just thought it should be a base feature. Also, I wasn't aware system.HasFocus was broken on non-Windows systems. Could you report that?

**Comment by iViscosity** [comment]
I run a Linux system and doesn't work for me. Also, I think it is:
```
Returns whether or not the game window has focus.

Returns true 100% of the time on OS X and returns nil on Linux.
```
From the [system.HasFocus() wiki page](http://wiki.garrysmod.com/page/system/HasFocus)

**Comment by Kefta** [comment]
Ah, didn't notice that. Thank you

**Comment by iViscosity** [comment]
Well once I test to see if it works maybe I can put a pull request.

**Comment by Kefta** [comment]
I believe this should be implemented in the engine since the order of hooks isn't guaranteed and the IN_ATTACK would still be registered. Also, you should probably be using CreateMove in your solution so the server would never know the difference of the user changing focus.
